### ISSUE TITLE
DB Import Doesn't Support SSL

### ISSUE THREAD

**Issue by michaelryanmcneill** [wp-cli/db-command#218]
# Bug Report
#### Summary
Running wp db import with the `--ssl` flag fails to properly connect to the database as expected. 

#### How to reproduce
To reproduce, simply attempt to import with a user that requires SSL. For example:
```
$ wp db import --ssl database-2022-01-24-a50feb0.sql
Error: Failed to get current SQL modes. Reason: ERROR 1045 (28000): Access denied for user 'dbuser'@'127.0.0.1' (using password: YES)
```
But, if you run `wp db cli --ssl`, you can successfully connect to the database:
```
$ wp db cli --ssl
Welcome to the MariaDB monitor.  Commands end with ; or \g.
Your MySQL connection id is 44
Server version: 8.0.27 Source distribution

Copyright (c) 2000, 2018, Oracle, MariaDB Corporation Ab and others.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

MySQL [database]>
```

#### Environment
```
OS:	Linux 5.11.0-1025-aws #27~20.04.1-Ubuntu SMP Fri Jan 7 13:09:56 UTC 2022 x86_64
Shell:	/bin/bash
PHP binary:	/RunCloud/Packages/php74rc/bin/php
PHP version:	7.4.26
php.ini used:	/etc/php74rc/php.ini
MySQL binary:	/usr/bin/mysql
MySQL version:	mysql  Ver 15.1 Distrib 10.5.13-MariaDB, for debian-linux-gnu (x86_64) using readline 5.2
SQL modes:
WP-CLI root dir:	phar://wp-cli.phar/vendor/wp-cli/wp-cli
WP-CLI vendor dir:	phar://wp-cli.phar/vendor
WP_CLI phar path:	/home/runcloud/webapps/example-com
WP-CLI packages dir:
WP-CLI global config:
WP-CLI project config:
WP-CLI version:	2.5.0
```

**Comment by AdsonCicilioti** [comment]
The `--http` flag don't help on this case?

(I ask because also don't know very well this scenario)

**Comment by michaelryanmcneill** [comment]
@AdsonCicilioti no, the `--http` flag will not help in this place. In this case the database is using TLS over TCP.

**Comment by danielbachhuber** [comment]
Thanks for the report, @michaelryanmcneill !

It looks like `wp db cli --ssl` works because the assoc args are passed directly through:

https://github.com/wp-cli/db-command/blob/196f4d3d171b79e19650182c645f854643439c9e/src/DB_Command.php#L382-L393

However, `wp db import` runs the arguments through some processing function:

https://github.com/wp-cli/db-command/blob/196f4d3d171b79e19650182c645f854643439c9e/src/DB_Command.php#L722-L734

It looks like this was added in https://github.com/wp-cli/db-command/pull/123

I think we'll need to adapt `::get_mysql_args()` to support `--ssl`.

**Comment by mmihelic13** [comment]
All `db` related commands listed on https://developer.wordpress.org/cli/commands/db/ would benefit from using the `--ssl` flag. As it stands some work over SSL and some do not.

**Comment by anthonyeden** [comment]
Hello,

I was just wondering what the path to resolving this is? Or if it's already been resolved?

Running ```db import``` on Alpine 3.21 with AWS RDS MySQL returns the following:

```
Failed to get current SQL modes. Reason: mysql: Deprecated program name. It will be removed in a future release, use '/usr/bin/mariadb' instead
TLS/SSL error: Certificate verification failure: The certificate is NOT trusted.
```

It looks like work is underway to fix the mysql/mariadb program name warning (https://github.com/wp-cli/db-command/issues/271 and https://github.com/wp-cli/db-command/commit/d44bf0d023ea840cc3a5df3a20f3106305b80b8f, but I'm unsure if https://github.com/wp-cli/wp-cli/blob/0187f2b4ce7e83f31bf4beecd5ad4f11ddf8b57b/php/utils.php#L1823 also needs work), but I'm not sure if anything has happened abut the SSL parameters?

**Comment by mickaelperrin** [comment]
And by extension, `--skip-ssl` isn't supported also and I now have issues running `wp db import` commands with MariaDB 10.6.21.

It looks like something changed in MariaDB configuration between 10.6.17 and 10.6.21.

**Comment by jfernandezr** [comment]
Hi, I'm in the same situation, I cannot use wp-cli in my CI scripts because of the following error:

```
Error: Failed to get current SQL modes. Reason: mysql: Deprecated program name. It will be removed in a future release, use '/usr/bin/mariadb' instead
ERROR 2026 (HY000): TLS/SSL error: self-signed certificate in certificate chain
```

Is there any temporary solution to fix this? I've tried disabling SSL in the ~/.my.cnf but it seems it's not used by wp-cli.

Thanks!

**Comment by Vilasamuni** [comment]
++metoo

**Comment by Vilasamuni** [comment]
Is this related to #271 ?

**Comment by kraftner** [comment]
I think this is related to #278
